pkg/eventmonitor/eventmonitor.go

```diff
@@ -158,6 +158,9 @@ func (m *EventMonitor) Start() error {
 
 // Close the module
 func (m *EventMonitor) Close() {
+	// stop so that consumers won't receive events anymore
+	m.Probe.Stop()
```

Comment:
There could be a scenario where the handler will get called after this function call, i.e. in case this function is called but the handler go routine is executing https://github.com/DataDog/datadog-agent/pull/18527/files#diff-2c30ea11397f5e442983afd45059898c30cefe144a266feafc8089fc79a82a98R373

Do we have a way of closing the perfbuffer/ringbuffer and thus make sure that the handler go routine stopped?